iron-spi-aws/src/main/java/io/axway/iron/spi/aws/s3/AwsS3SnapshotStore.java

```diff
@@ -104,7 +119,22 @@ public InputStream inputStream() {
 
     private List<S3ObjectSummary> listSnapshotFilesForId(BigInteger transactionId) {
         String snapshotDirectoryName = getSnapshotDirectoryName(transactionId);
-        return m_amazonS3.listObjectsV2(m_bucketName, snapshotDirectoryName).getObjectSummaries();
+        ListObjectsV2Request listObjectsRequest = new ListObjectsV2Request().withBucketName(m_bucketName).withPrefix(snapshotDirectoryName);
+        return listAllObjects(listObjectsRequest, ListObjectsV2Result::getObjectSummaries);
+    }
+
+    @Nonnull
+    private <T> List<T> listAllObjects(ListObjectsV2Request listObjectsRequest, Function<ListObjectsV2Result, List<T>> listObjectsResultMapper) {
```

Comment:
Could have use Publisher, but not simple with aws-java-sdk V1. Will be easier with awssdk V2.